### PR TITLE
rta: supernode announce handler and sender

### PR DESCRIPTION
**Pull request by mbg033**
- handling incoming announces (/dapi/v2.0/send_supernode_announce);
- periodically posting announce (/json_rpc/rta/)

### CODE REVIEW

File: data/config.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [cryptonode]
-rpc-address=127.0.0.1:18981
+rpc-address=127.0.0.1:28881
```
Comment (EDDragonWolf) [inline data/config.ini]: Maybe it is bad idea to change port

File: src/graft_manager.cpp [hunk 2]
```diff
@@ -351,7 +351,7 @@ void BaseTask::processResult()
     }
 }
 
-void BaseTask::onCryptonDone(CryptoNodeSender &cns)
+    void BaseTask::onCryptonDone(CryptoNodeSender &cns)
```
Comment (EDDragonWolf) [inline src/graft_manager.cpp]: Why do you add this spaces?

File: src/main.cpp [hunk 3]
```diff
@@ -36,6 +55,56 @@ void addGlobalCtxCleaner(graft::Manager& manager, int ms)
                 );
 }
 
+
+void startSupernodePeriodicTasks(graft::Manager& manager, size_t interval_ms)
```
Comment (EDDragonWolf) [inline src/main.cpp]: I do not think that it's the best idea keep this method in `main.cpp`, maybe better is to move it into the separated file.

File: src/main.cpp [hunk 4]
```diff
@@ -112,12 +198,24 @@ int main(int argc, const char** argv)
...
+            sopts.data_dir = p.string();
+        }
+
         int lru_timeout_ms = server_conf.get<int>("lru-timeout-ms");
+        std::string stake_wallet_name = server_conf.get<string>("stake-wallet-name", "stake-wallet");
 
         const boost::property_tree::ptree& cryptonode_conf = config.get_child("cryptonode");
         sopts.cryptonode_rpc_address = cryptonode_conf.get<string>("rpc-address");
-        //sopts.cryptonode_p2p_address = cryptonode_conf.get<string>("p2p-address");
+        // sopts.cryptonode_p2p_address = cryptonode_conf.get<string>("p2p-address");
```
Comment (EDDragonWolf) [inline src/main.cpp]: Are we still need this?

File: src/main.cpp [hunk 5]
```diff
@@ -127,12 +225,60 @@ int main(int argc, const char** argv)
...
+
+        std::cout << boost::filesystem::absolute(data_path).string() << std::endl;
+        sopts.watchonly_wallets_path = watchonly_wallets_path.string();
+        graft::Manager manager(sopts);
         addGlobalCtxCleaner(manager, lru_timeout_ms);
 
+        // create supernode instance and put it into global context
+        graft::FullSupernodeList::SupernodePtr supernode {new graft::Supernode(
+                        (stake_wallet_path / stake_wallet_name).string(),
+                        "", // TODO
```
Comment (EDDragonWolf) [inline src/main.cpp]: What TODO?

File: src/routers_http.cpp [hunk 6]
```diff
@@ -13,15 +13,15 @@ namespace graft {
...
-        Router http_router;
-        graft::registerRTARequests(http_router);
-        m.addRouter(http_router);
+        // dapi_router.addRoute("/test", METHOD_GET, h3_test1);
+        graft::registerRTARequests(dapi_router);
 
+        m.addRouter(dapi_router);
+        // Router http_router;
+        // graft::registerRTARequests(http_router);
+        // m.addRouter(http_router);
```
Comment (EDDragonWolf) [inline src/routers_http.cpp]: Why this code commented?

**Comment by mbg033** [comment]
Closing, this functionality is in more recent PR https://github.com/graft-project/graft-ng/pull/76

**Pull request closed without merging**